adala/runtimes/_litellm.py

```diff
@@ -0,0 +1,488 @@
+
+        outputs = {}
+        for output_field in output_fields:
+            if output_field['type'] == 'text':
+                if user_prompt is not None:
+                    user_prompt += f"\n{output_field['text']}"
+                else:
+                    user_prompt = output_field['text']
+            elif output_field['type'] == 'var':
+                name = output_field['text']
+                messages.append({'role': 'user', 'content': user_prompt})
+                completion_text = self.execute(messages)
+                if name in options:
+                    completion_text = match_options(
+                        completion_text, options[name]
+                    )
+                outputs[name] = completion_text
+                messages.append(
+                    {'role': 'assistant', 'content': completion_text}
+                )
+                user_prompt = None
+
+        return outputs
+
+
+class AsyncLiteLLMChatRuntime(AsyncRuntime):
+    """
+    Runtime that uses [OpenAI API](https://openai.com/) and chat completion models to perform the skill.
+    It uses async calls to OpenAI API.
+
+    Attributes:
+        model: OpenAI model name.
+        openai_api_key: OpenAI API key. If not provided, will be taken from OPENAI_API_KEY environment variable.
+        base_url: Can point to any implementation of the OpenAI API. Defaults to OpenAI's.
+        max_tokens: Maximum number of tokens to generate. Defaults to 1000.
+        temperature: Temperature for sampling, between 0 and 1. Higher values means the model will take more risks.
+            Try 0.9 for more creative applications, and 0 (argmax sampling) for ones with a well-defined answer.
+            Defaults to 0.0.
+
+        concurrent_clients: Number of concurrent clients to OpenAI API. More clients means more parallel requests, but
+            also more money spent and more chances to hit the rate limit. Defaults to 10.
+    """
+
+    model_config = ConfigDict(
+        arbitrary_types_allowed=True
+    )  # for @computed_field
+
+    model: str
+    api_key: Optional[str] = None
+    base_url: Optional[str] = None
+    max_tokens: Optional[int] = 1000
+    temperature: Optional[float] = 0.0
+    splitter: Optional[str] = None
+    timeout: Optional[int] = 10
+
+    def _init_http_client(self) -> httpx.AsyncClient:
+        client = (
+            httpx.AsyncClient(
+                timeout=self.timeout,
+            ),
+        )
+        litellm.aclient_session = client
+        return client
+
+    def init_runtime(self) -> 'Runtime':
+        # check model availability
+        try:
+            if self.api_key:
+                litellm.check_valid_key(model=self.model, api_key=self.api_key)
+        except NotFoundError:
+            raise ValueError(
+                f'Requested model "{self.model}" is not available in your OpenAI account.'
+            )
+        self._http_client = self._init_http_client
```

Comment:
```suggestion
        self._http_client = self._init_http_client()
```
missing parens?

Reply:
also is there a reason this isn't a `@property` like it was in the openai runtimes?

Reply:
Yeah how did it work at all with missing parens?

Reply:
huh, yeah that is weird, I have definitely been running this locally...